interface/src/app/tribe_client/genesets.js

```diff
@@ -0,0 +1,97 @@
+angular.module('adage.tribe_client.genesets', [
+  'adage.tribe_client.resource'
+])
+
+// Geneset Search Factory
+// Allows for retreiving for and interacting with search results for genesets.
+.factory('GenesetSearch', ['Genesets', function(Genesets) {
+  var genesets = [];
```

Comment:
Wait a second.. I may have a simplification for you here.

The trouble we're having is from passing around copies of `genesets`, right? The *Angularish* solution should be to simply reference the One True `genesets` wherever you need to, and two-way binding will handle the rest.

The problem here is that the One True `genesets` is locked up in this closure variable inside the function that implements the `GenesetSearch` service. What if, instead, these `var`s were moved into the object that gets returned (line 12 below)? Then you can bind `GenesetSearch.genesets` directly wherever it's needed.

For an example of how I've done that, refer to the `SampleBin` implementation in `interface/src/app/analysis/sampleBin.js`.